pop-api/src/v0/assets/trust_backed.rs

```diff
@@ -0,0 +1,519 @@
+/// 25.	set_min_balance
+/// 26. touch_other
+/// 27.	refund_other
+/// 28. block
+
+
+/// Issue a new class of fungible assets from a public origin.
+pub fn create(
+	id: AssetId,
+	admin: impl Into<MultiAddress<AccountId, ()>>,
+	min_balance: Balance,
+) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::Create {
+		id: id.into(),
+		admin: admin.into(),
+		min_balance: Compact(min_balance),
+	}))?)
+}
+
+/// Start the process of destroying a fungible asset class.
+pub fn start_destroy(id: AssetId) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::StartDestroy {
+		id: id.into(),
+	}))?)
+}
+
+/// Destroy all accounts associated with a given asset.
+pub fn destroy_accounts(id: AssetId) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::DestroyAccounts {
+		id: id.into(),
+	}))?)
+}
+
+/// Destroy all approvals associated with a given asset up to the max (see runtime configuration TrustBackedAssets `RemoveItemsLimit`).
+pub fn destroy_approvals(id: AssetId) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::DestroyApprovals {
+		id: id.into(),
+	}))?)
+}
+
+/// Complete destroying asset and unreserve currency.
+pub fn finish_destroy(id: AssetId) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::FinishDestroy {
+		id: id.into(),
+	}))?)
+}
+
+/// Mint assets of a particular class.
+pub fn mint(
+	id: AssetId,
+	beneficiary: impl Into<MultiAddress<AccountId, ()>>,
+	amount: Balance,
+) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::Mint {
+		id: id.into(),
+		beneficiary: beneficiary.into(),
+		amount: Compact(amount),
+	}))?)
+}
+
+/// Reduce the balance of `who` by as much as possible up to `amount` assets of `id`.
+pub fn burn(
+	id: AssetId,
+	who: impl Into<MultiAddress<AccountId, ()>>,
+	amount: Balance,
+) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::Burn {
+		id: id.into(),
+		who: who.into(),
+		amount: Compact(amount),
+	}))?)
+}
+
+/// Move some assets from the sender account to another.
+pub fn transfer(
+	id: AssetId,
+	target: impl Into<MultiAddress<AccountId, ()>>,
+	amount: Balance,
+) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::Transfer {
+		id: id.into(),
+		target: target.into(),
+		amount: Compact(amount),
+	}))?)
+}
+
+/// Move some assets from the sender account to another, keeping the sender account alive.
+pub fn transfer_keep_alive(
+	id: AssetId,
+	target: impl Into<MultiAddress<AccountId, ()>>,
+	amount: Balance,
+) -> Result<()> {
+	Ok(dispatch(RuntimeCall::TrustBackedAssets(TrustBackedAssetsCalls::TransferKeepAlive {
+		id: id.into(),
+		target: target.into(),
+		amount: Compact(amount),
+	}))?)
+}
+
+/// Move some assets from one account to another. Sender should be the Admin of the asset `id`.
+pub fn force_transfer(
```

Comment:
`force_X` functions should not be in the API as they require root origin.

Reply:
I don't think this is technically true for this pallet, see comment above the function.

Reply:
Correct, some of the `force_X` functions allow other origins than `ForceOrigin` from the config. E.g. for this case it is the admin of the asset id, which can be a smart contract

Reply:
Apologies. I see now. `force_create` is the only one